### PR TITLE
fix(db): check `DownloadedAt` for `trivy-java-db`

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
in addition to checking `NextUpdate` - also check that `trivy-java-db` was downloaded more than one day ago (check `DownloadedAt` field).

## Related Discussions
- https://github.com/aquasecurity/trivy/discussions/7591#discussioncomment-10748187

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Review by knqyf263 — APPROVED** [review]

File: pkg/javadb/client.go [hunk 1]
```diff
@@ -85,6 +85,20 @@ func (u *Updater) Update() error {
...
 
+func (u *Updater) isNewDB(meta db.Metadata) bool {
+	now := time.Now().UTC()
+	if now.Before(meta.NextUpdate) {
+		log.Debug("Java DB update was skipped because the local Java DB is the latest")
+		return true
+	}
+
+	if now.Before(meta.DownloadedAt.Add(time.Hour * 24)) { // 1 day
+		log.Debug("Java DB update was skipped because the local Java DB was downloaded during the last hour")
```
Comment (sakky016) [inline pkg/javadb/client.go:96]: Log message should be updated, (day, not hour)

Comment (knqyf263) [reply pkg/javadb/client.go:96]: Nice catch. I overlooked it. Thanks!

Comment (DmitriyLewen) [reply pkg/javadb/client.go]: Thanks!
Updated in https://github.com/aquasecurity/trivy/pull/7592/commits/1de0a8177759c428dc7e371d9b467c310eb00dc8

**Comment by Amejonah1200** [comment]
But isn't the db updated like 2 or more times a day?

**Comment by DmitriyLewen** [comment]
@Amejonah1200 
`Trivy-java-db` is updated once a day:
https://github.com/aquasecurity/trivy-java-db/blob/c46eaa15cf3669a3dde904991b69359ff0cd23ee/.github/workflows/cron.yml#L5

You might have thought of `trivy-db` (we update `trivy-db` every 6 hours)

**Pull request merged**